src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -244,6 +253,44 @@ final case class AgreementsApiServiceImpl(
       eService                         <- catalogManagementService.getEService(agreement.eserviceId)
     } yield (producerTenant.name, consumerTenant.name, consumerTenant, eService)
 
+  def parallelGetTenantsService(agreement: AgreementProcess.Agreement)(implicit
+    contexts: Seq[(String, String)]
+  ): Future[(TenantManagement.Tenant, TenantManagement.Tenant, CatalogManagement.EService)] =
+    Applicative[Future].tuple3(
+      tenantManagementService
+        .getTenant(agreement.consumerId),
+      tenantManagementService.getTenant(agreement.producerId),
+      catalogManagementService.getEService(agreement.eserviceId)
+    )
```

Comment:
Let's reuse the existing `parallelGet` instead of creating a new function.
I suggest to change the original in order to return `(consumerTenant, producerTenant, eService)` exactly like this one.

FYI @TonioGela found that it is not recommended to use Cats functions on Future because it is not clear how it handles the execution context and the parallelism.

Reply:
I endorse what @galales said. First of all, I'm happy that you've tried to write such a function, it's a not that trivial task and you've quite succeeded, but I have a couple of suggestions for you.

In particular:
- Do not invoke cats typeclasses directly (i.e. `Applicative[Future].tuple3`) but do rely on syntax (i.e. `(fa,fb,fc).tupled` or similar)
- Do not use at all cats methods with `Futures` (i.e. prefer `Future.traverse(list)(f)` instead of `list.traverse(f)`)

For this particular case I suggest using `Future` combinators directly, I'll do something like 

```scala
tenantManagementService.getTenant(agreement.consumerId)
  .zipWith(tenantManagementService.getTenant(agreement.producerId))(identity)
   .zipWith(catalogManagementService.getEService(agreement.eserviceId)){case ((a,b),c) => (a,b,c)}
```

it's immensely uglier, but it's safer, as `zipWith` requires an `ExecutionContext` on whom to wait for the 3 `Future`s for example.
(Consider that I wrote the code w/o an IDE, it might not compile, but feel free to ask for help in case it's needed)